dlt/destinations/job_client_impl.py

```diff
@@ -113,6 +119,99 @@ def is_sql_job(file_path: str) -> bool:
         return os.path.splitext(file_path)[1][1:] == "sql"
 
 
+class ModelLoadJob(RunnableLoadJob, HasFollowupJobs):
+    """
+    A job to insert rows into a table from a model file which contains a single select statement
+    """
+
+    def __init__(self, file_path: str) -> None:
+        super().__init__(file_path)
+        self._job_client: "SqlJobClientBase" = None
+
+    def run(self) -> None:
+        with FileStorage.open_zipsafe_ro(self._file_path, "r", encoding="utf-8") as f:
+            select_dialect = f.readline().split(": ")[1].strip() or None
+            select_statement = f.read()
+
+        sql_client = self._job_client.sql_client
+        insert_statement = self._insert_statement_from_select_statement(
+            select_dialect, select_statement
+        )
+        sql_client.execute_sql(insert_statement)
+
+    def _insert_statement_from_select_statement(
+        self, select_dialect: Optional[str], select_statement: str
+    ) -> str:
+        """
+        Generates an INSERT statement from a SELECT statement using sqlglot.
+        Transpiles the SELECT if needed and adjusts identifiers for the destination dialect.
+        NOTE: Here we generate an insert statement from a select statement, this is the duckdb
+        dialect, we may be able to transpile with sqlglot for each destination or we need
+        to subclass and override this method.
+        """
+        sql_client = self._job_client.sql_client
+        target_table = sql_client.make_qualified_table_name(self._load_table["name"])
+        target_catalog = sql_client.catalog_name(escape=False)
+
+        # Adjust separator for clickhouse-to-other
+        if (
+            self._job_client.config.destination_type != "clickhouse"
+            and select_dialect == "clickhouse"
+        ):
+            select_statement = select_statement.replace("___", '"."')
+
+        # Parse SELECT
+        parsed_select_query = sqlglot.parse_one(select_statement, read=select_dialect)
+
+        # Adjust table parts (catalog/db/this) based on dialect and catalog presence
+        for table in parsed_select_query.find_all(sqlglot.exp.Table):
+            parts = list(table.parts)
+            if target_catalog:
+                if len(parts) == 3:
+                    table.set("catalog", sqlglot.exp.to_identifier(target_catalog))
```

Comment:
IMO if someone set the catalog we must keep it. it could be some kind of federated database